src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -1577,14 +1609,13 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 	}
 
 	private doLayout(dimensions: IEditorTitleControlDimensions, options?: IMultiEditorTabsControlLayoutOptions): void {
-
-		// Only layout if we have valid tab index and dimensions
-		const activeTabAndIndex = this.group.activeEditor ? this.getTabAndIndex(this.group.activeEditor) : undefined;
-		if (activeTabAndIndex && dimensions.container !== Dimension.None && dimensions.available !== Dimension.None) {
-
+		// Only layout when there is an active editor
+		// The active editor might be located in a different tab row
+		if (this.groupViewer.activeEditor && dimensions.container !== Dimension.None && dimensions.available !== Dimension.None) {
 			// Tabs
-			const [activeTab, activeIndex] = activeTabAndIndex;
-			this.doLayoutTabs(activeTab, activeIndex, dimensions, options);
+			const activeTabAndIndex = this.tabsModel.activeEditor ? this.getTabAndIndex(this.tabsModel.activeEditor) : undefined;
```

Comment:
I would move this code into the `doLayoutTabsNonWrapping` method where it is being used to clean up things.